File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/AccountBalance.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.TimestampDeserializer;
import com.umbrella.insurance.core.serializers.TimestampSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "account_balances", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "account_balances_unique", columnNames = {"user_id", "account_balance_type_id", "unit_id"})
})
public class AccountBalance {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "account_balance_id", nullable = false)
    private Long id;

    @NotNull
    @Column(name = "account_balance_value", nullable = false)
    private Double accountBalanceValue;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "account_balance_type_id")
    private AccountBalanceType accountBalanceType;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @NotNull
    @Column(name = "updated_date_time", nullable = false)
    private Timestamp updatedDateTime;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "unit_id")
    private Unit unit;

}

File: umbrella-insurance-react-ui/src/models/users/accountBalances/v1/AccountBalance.ts
import { Unit } from "../../../units/v1/Unit";
import { User } from "../../v1/User";
import { AccountBalanceType } from "../accountBalanceTypes/v1/AccountBalanceType";

export class AccountBalance {
    id?: number;
    accountBalanceValue?: number;
    accountBalanceType?: AccountBalanceType;
    updatedDateTime?: string;
    user?: User;
    unit?: Unit;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.accountBalanceValue) {
                this.accountBalanceValue = obj.accountBalanceValue;
            }
            if(obj?.accountBalanceType) {
                this.accountBalanceType = new AccountBalanceType(obj.accountBalanceType);
            }
            if(obj?.updatedDateTime) {
                this.updatedDateTime = obj.updatedDateTime;
            }
            if(obj?.user) {
                this.user = new User(obj.user);
            }
            if(obj?.unit) {
                this.unit = new Unit(obj.unit);
            }
        }
    }
}